pkgs/desktops/deepin/core/dde-dock/default.nix

```diff
@@ -0,0 +1,91 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, dtkwidget
+, dde-qt-dbus-factory
+, qt5integration
+, qt5platform-plugins
+, dde-control-center
+, deepin-desktop-schemas
+, cmake
+, qttools
+, qtx11extras
+, pkg-config
+, wrapQtAppsHook
+, wrapGAppsHook
+, gsettings-qt
+, libdbusmenu
+, xorg
+, glib
+, gtest
+, qtbase
+}:
+
+stdenv.mkDerivation rec {
+  pname = "dde-dock";
+  version = "5.5.81";
+
+  src = fetchFromGitHub {
+    owner = "linuxdeepin";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-x8U5QPfIykaQLjwbErZiYbZC+JyPQQ+jd6MBjDQyUjs=";
+  };
+
+  postPatch = ''
+    substituteInPlace plugins/tray/system-trays/systemtrayscontroller.cpp frame/controller/dockpluginscontroller.cpp \
+      --replace "/usr/lib/dde-dock/plugins" "/run/current-system/sw/lib/dde-dock/plugins"
+
+    substituteInPlace plugins/show-desktop/showdesktopplugin.cpp frame/window/components/desktop_widget.cpp \
+      --replace "/usr/lib/deepin-daemon" "/run/current-system/sw/lib/deepin-daemon"
+
+    substituteInPlace plugins/{dcc-dock-plugin/settings_module.cpp,tray/system-trays/systemtrayscontroller.cpp} \
+      --replace "/usr" "$out"
+    '';
+
+  nativeBuildInputs = [
+    cmake
+    qttools
+    pkg-config
+    wrapQtAppsHook
+    wrapGAppsHook
+  ];
+  dontWrapGApps = true;
+
+  buildInputs = [
+    dtkwidget
+    qt5platform-plugins
+    dde-qt-dbus-factory
+    dde-control-center
+    deepin-desktop-schemas
+    qtx11extras
+    gsettings-qt
+    libdbusmenu
+    xorg.libXcursor
+    xorg.libXtst
+    xorg.libXdmcp
+    gtest
+  ];
+
+  outputs = [ "out" "dev" ];
+
+  cmakeFlags = [ "-DVERSION=${version}" ];
+
+  # qt5integration must be placed before qtsvg in QT_PLUGIN_PATH
+  qtWrapperArgs = [
+    "--prefix QT_PLUGIN_PATH : ${qt5integration}/${qtbase.qtPluginPrefix}"
+  ];
+
+  preFixup = ''
+    glib-compile-schemas ${glib.makeSchemaPath "$out" "${pname}-${version}"}
```

Comment:
Isn't this done automatically? Should we create a hook if not?

Reply:
For desktop environment，we need some like：
https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/gnome/nixos/gsettings-overrides/default.nix

glib-compile-schemas for single application maybe not necessary